compiler/semexprs.nim

```diff
@@ -1419,6 +1419,12 @@ proc dotTransformation(c: PContext, n: PNode): PNode =
     result = newNodeI(nkDotCall, n.info)
     result.add n[1]
     result.add copyTree(n[0])
+  elif n[0].typ != nil and n[0].typ.kind == tyError:
+    # XXX: this hack avoids the issue where we keep reevaluating the same
```

Comment:
a test linked to here is worth a thousand words (but ok to explain too)

Reply:
It's t11166.nim that's demonstrates the issue, the reason I didn't reference it in the comment is because I believe (could be wrong) that the issue should be fixed elsewhere.

Reply:
still, it helps to add refs t11166.nim in the comment